src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceMappingOptionsControl.xaml.cs

```diff
@@ -0,0 +1,250 @@
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        //converts from list of packagesourcemappingsourceItems to a dictonary that can be read by UI
+        private IReadOnlyList<MappingUIDisplay> ReadMappingsFromConfigToUI(IReadOnlyList<PackageSourceMappingSourceItem> originalMappings)
+        {
+            Dictionary<string, List<PackageSourceContextInfo>> uiSourceMappings = new Dictionary<string, List<PackageSourceContextInfo>>();
+            foreach (PackageSourceMappingSourceItem sourceItem in originalMappings)
+            {
+                foreach (PackagePatternItem patternItem in sourceItem.Patterns)
+                {
+                    if (!uiSourceMappings.ContainsKey(patternItem.Pattern))
+                    {
+                        uiSourceMappings[patternItem.Pattern] = new List<PackageSourceContextInfo>();
+                    }
+                    uiSourceMappings[patternItem.Pattern].Add(new PackageSourceContextInfo(sourceItem.Key));
+                }
+            }
+            List<MappingUIDisplay> mappingsCollection = new List<MappingUIDisplay>();
+            foreach (string packageID in uiSourceMappings.Keys)
+            {
+                MappingUIDisplay temp = new MappingUIDisplay(packageID, uiSourceMappings[packageID]);
+                mappingsCollection.Add(temp);
+            }
+            return mappingsCollection.AsReadOnly();
+        }
+
+        //converts from dictonary created by UI to list of packageSourceMappingSourceItems
+        private IReadOnlyList<PackageSourceMappingSourceItem> ReadMappingsFromUIToConfig(ItemsChangeObservableCollection<MappingUIDisplay> uiSourceMappings)
+        {
+            Dictionary<string, List<PackagePatternItem>> mappingsDictonary = new Dictionary<string, List<PackagePatternItem>>();
+            foreach (MappingUIDisplay mappingUIDisplay in uiSourceMappings)
+            {
+                foreach (PackageSourceContextInfo source in mappingUIDisplay.Sources)
+                {
+                    //Contains method did not work since diff instances of MappingUIDisplay even though name is the same
+                    //made own contains method
+                    bool newSource = true;
+                    foreach (string mapping in mappingsDictonary.Keys)
+                    {
+                        if (mapping == source.Name)
+                        {
+                            newSource = false;
+                        }
+                    }
+                    if (newSource == true)
+                    {
+                        mappingsDictonary[source.Name] = new List<PackagePatternItem>();
+                    }
+                    PackagePatternItem tempID = new PackagePatternItem(mappingUIDisplay.ID);
+                    bool newID = true;
```

Comment:
I believe line `224-236` can be replaced with below, which more readable.

 ```
                   PackagePatternItem tempID = new PackagePatternItem(mappingUIDisplay.ID);

                    if (!mappingsDictonary[source.Name].Any(id => id.Pattern == tempID.Pattern))
                    {
                        mappingsDictonary[source.Name].Add(tempID);
                    }
```

Reply:
will do. thanks